server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -240,6 +241,9 @@ public TabletServerMinCMetrics getMinCMetrics() {
   private final WalStateManager walMarker;
   private final ServerContext context;
 
+  private int maxThreadPermits = 0;
+  private Semaphore sem;
```

Comment:
If using optional, could make this instance var optiona so that we can avoid having to recreated the optional object.

```suggestion
  private Optional<Semaphore> sem;
```